targets/arm/mikroe/sam/src/uart/implementations/implementation_1/hal_ll_uart.c

```diff
@@ -0,0 +1,1165 @@
+#define hal_ll_uart_get_baud_error(_baud_real,_baud) (((float)(abs(_baud_real - _baud)) / _baud) * 100)
+
+/*!< @brief Macros defining bit location */
+#define HAL_LL_PID_UART0_BIT        7
+#define HAL_LL_PID_UART1_BIT        8
+#define HAL_LL_PID_UART2_BIT        12
+#define HAL_LL_PID_UART3_BIT        13
+#define HAL_LL_PID_UART4_BIT        14
+
+#define HAL_LL_PID_USART0_BIT       13
+#define HAL_LL_PID_USART1_BIT       14
+#define HAL_LL_PID_USART2_BIT       15
+
+#define HAL_LL_CR_RXEN_BIT          4
+#define HAL_LL_CR_RXDIS_BIT         5
+#define HAL_LL_CR_TXEN_BIT          6
+#define HAL_LL_CR_TXDIS_BIT         7
+
+#define HAL_LL_UART_IER_RXRDY_BIT   0
+#define HAL_LL_UART_IER_TXRDY_BIT   1
+
+#define HAL_LL_UART_IMR_RXRDY_FLAG 0x1
+#define HAL_LL_UART_IMR_TXRDY_FLAG 0x2
+
+#define HAL_LL_UART_SR_RXRDY_FLAG 0x1
+#define HAL_LL_UART_SR_TXRDY_FLAG 0x2
+
+/*!< @#brief Macros defining register masks */
+#define HAL_LL_UART_MR_DATA_LENGTH_7_BIT_MASK       0x0080
+#define HAL_LL_UART_MR_DATA_LENGTH_8_BIT_MASK       0x00C0
+#define HAL_LL_UART_MR_PARITY_EVEN_MASK             0x0000
+#define HAL_LL_UART_MR_PARITY_ODD_MASK              0x0200
+#define HAL_LL_UART_MR_PARITY_NONE_MASK             0x0800
+#define HAL_LL_UART_MR_STOP_BITS_ONE_MASK           0x0000
+#define HAL_LL_UART_MR_STOP_BITS_ONE_AND_HALF_MASK  0x1000
+#define HAL_LL_UART_MR_STOP_BITS_TWO_MASK           0x2000
+
+#define HAL_LL_UART_CR_DISABLE_MASK                 0x00AC
+
+/*!< @brief Helper macro for getting hal_ll_module_state address */
+#define hal_ll_uart_get_module_state_address ((hal_ll_uart_handle_register_t *)*handle)
+
+/*!< @brief Macro used for interrupt status register flag check
+ * Used in interrupt handlers.
+ */
+#define hal_ll_uart_get_status_flags(_handle,_flag) (((hal_ll_uart_base_handle_t *)_handle)->imr & _flag)
+
+/*!< @brief Macro used for module interrupt source check.
+ * Used in interrupt handlers.
+ */
+ #define hal_ll_uart_get_interrupt_source(_handle,_bit) (((hal_ll_uart_base_handle_t *)_handle)->sr & _bit)
+
+/*!< @brief UART HW register structure */
+typedef struct
+{
+    hal_ll_base_addr_t cr;
+    hal_ll_base_addr_t mr;
+    hal_ll_base_addr_t ier;
+    hal_ll_base_addr_t idr;
+    hal_ll_base_addr_t imr;
+    hal_ll_base_addr_t sr;
+    hal_ll_base_addr_t rhr;
+    hal_ll_base_addr_t thr;
+    hal_ll_base_addr_t brgr;
+    hal_ll_base_addr_t cmpr;
+} hal_ll_uart_base_handle_t;
+
+/*!< @brief UART baud rate structure */
+typedef struct
+{
+    uint32_t baud;
+    uint32_t real_baud;
+} hal_ll_uart_baud_t;
+
+/*!< @brief UART hw specific structure */
+typedef struct
+{
+    hal_ll_base_addr_t base;
+    hal_ll_pin_name_t module_index;
+    hal_ll_uart_pins_t pins;
+    hal_ll_uart_baud_t baud_rate;
+    hal_ll_uart_parity_t parity;
+    hal_ll_uart_stop_bits_t stop_bit;
+    hal_ll_uart_data_bits_t data_bit;
+} hal_ll_uart_hw_specifics_map_t;
+
+/*!< @brief UART hw specific module values */
+typedef struct
+{
+    uint16_t pin_tx;
```

Comment:
Check file for appropriate use of `hal_ll_base_addr_t` and `hal_ll_pin_name_t`.

Reply:
Changed it for all source files where needed.